se_core/include/se/node_iterator.hpp

```diff
@@ -166,33 +175,45 @@ class node_iterator {
     vec3i frontierVoxels;
     frontierVoxels.clear();
 
-    Eigen::Vector3i blockCoord = unpack_morton(morton);
-    int xlast = blockCoord(0) + BLOCK_SIDE;
-    int ylast = blockCoord(1) + BLOCK_SIDE;
-    int zlast = blockCoord(2) + BLOCK_SIDE;
+    typename VoxelBlock<T>::value_type value;
+    const Eigen::Vector3i blockCoord = keyops::decode(morton);
+    const int level = keyops::level(morton);
+
+    Node<T> *node = nullptr;
+    bool is_block = false;
+    map_.fetch_octant(blockCoord(0), blockCoord(1), blockCoord(2), node, is_block);
+    if (!is_block) {
```

Comment:
It would be easier to follow if the if and else were swapped so that the test is `if (is_block)`.